### ISSUE TITLE
Use of library from commonMain in Kotlin Multiplatform Project

### ISSUE THREAD

**Issue by joreilly** [awslabs/aws-sdk-kotlin#460]
### Describe the issue

I can include following for example as `commonMain` dependency in Kotlin Multiplatform project and access/use code from JVM and Android code but if I try to use from `commonMain` source then only certain symbols appear to be resolvable (for example can access `StaticCredentialsProvider` but not `DynamoDbClient`).  Perhaps this is expected right now?

```
api("aws.sdk.kotlin:dynamodb:0.9.4-beta")
```

### Steps to Reproduce

per description

### Current behavior

per description

### AWS Kotlin SDK version used

0.9.4

### Platform (JVM/JS/Native)

JVM/Android/Common

### Operating System and version

macOS 11.4

**Comment by aajtodd** [comment]
Yes this is expected right now. The services are built with the Kotlin/JVM [plugin](https://github.com/awslabs/aws-sdk-kotlin/blob/main/services/build.gradle.kts#L24) still. You should be able to access them from a JVM sourceSet. Most of the runtime itself is implemented with the [multiplatform plugin](https://github.com/awslabs/aws-sdk-kotlin/blob/main/aws-runtime/build.gradle.kts#L45) which is why you can access it from `common`. 

Support for other targets is on our roadmap. The best thing you can do right now is to upvote the existing issues with the targets you would like to see supported. 

I agree that the ability to use the services in a `common` sourceSet (targeting only `jvm()` and `android()` for now of course) would be useful but it may not come until we get to supporting one of the other KMP targets outright.

**Comment by kggilmer** [comment]
Here is a list of all of our existing feature requests @joreilly , and these two are particular to [KMP/js](https://github.com/awslabs/aws-sdk-kotlin/issues/228) and [KMP/native](https://github.com/awslabs/aws-sdk-kotlin/issues/229) support.

**Comment by ianbotsf** [comment]
The fix for this issue has been pushed to **main** and should be included in the next release (likely this week). With that release, it should be possible to reference service clients and related types from **common** source sets (instead of requiring JVM).

Note that the AWS SDK for Kotlin does yet not support non-JVM targets but they are on our roadmap as #228 and #229.

**Comment by joreilly** [comment]
I see that a 0.16.0 version was released but doesn't look like it's been published yet?

**Comment by aajtodd** [comment]
We were having some issues syncing with maven central, it appears to be available now (e.g. [dynamodb-jvm](https://repo1.maven.org/maven2/aws/sdk/kotlin/dynamodb-jvm/))

NOTE: we accidentally cut this release without the `-beta` qualifier. Future dev preview releases will use this qualifier again though.

**Comment by joreilly** [comment]
Let me know if it's better to create another issue for this but getting following now when trying (in JVM code) to populate DynamoDb table (on M1 device)....seems related to https://github.com/aws/aws-sdk-java-v2/issues/2942 ?

```
Unable to unpack AWS CRT lib: java.io.IOException: Unable to open library in jar for AWS CRT: /osx/armv8/libaws-crt-jni.dylib
java.io.IOException: Unable to open library in jar for AWS CRT: /osx/armv8/libaws-crt-jni.dylib
```

**Comment by ianbotsf** [comment]
That's correct @joreilly, this is a known issue. We're in the final stages of changes which will remove the CRT as a required dependency and enable the AWS SDK for Kotlin to run on Apple M1 so hopefully that issue won't be around much longer.
